main.tex

```diff
@@ -88,11 +88,11 @@ \subsubsection{Minimum denomination}
     \label{fig:minimumdenomination}
 \end{figure}
 
-Even if users are able to provide several smaller value inputs with total value greater than the minimum denomination, the coordinator knows those inputs belong to the same user. In an ideal mixing protocol the coordinator should not obtain more information than the already available blockchain data by coordinating the CoinJoin transaction. This is especially important in practice as identifying  sub-transactions when many multiple inputs and outputs are provided by the same parties is often computationally expensive with currently known algorithms, even without equal amounts being utilized~\cite{maurer2017anonymous}.
+Even when users are able to provide several smaller value inputs with total value greater than the minimum denomination, the coordinator knows those inputs belong to the same user. In an ideal mixing protocol the coordinator should not obtain more information than the already available public ledger data by coordinating the CoinJoin transaction. This is especially important in practice as identifying sub-transactions when many multiple inputs and outputs are provided by the same parties is often computationally expensive using currently known algorithms, even without equal amounts being utilized~\cite{maurer2017anonymous}.
```

Comment:
this is not related to the change, but i thought i might open it up for discussion anyway

i think the computational complexity thing is a red herring, same as my criticism of cashfusion and same as the sharedcoin weaknesses: solving special cases of these problems is not so hard, especially when you care about constrained sub-txns. secondly, the attack gets exponentially easier as individual targets are de-anonymized. i don't think that's a valid hardness assumption and therefore it's not related.

what the knapsack paper does is more subtle, by construction it guarantees a minimum number of valid solutions that can be trivially found. instead of making it hard for the adversary to find one solution, finding solutions is easy, the problem is deciding between them without relying on external information

Reply:
what is a sub-transaction in this line? does it mean one of multiple inputs and outputs provided by the same party? or one of the transactions of the protocol as a whole. Sorry, I got lost here

Reply:
the first one, defined in the knapsack paper (`maurer2017anonymous` citation)

Reply:
```suggestion
Even when users are able to provide several smaller value inputs with total value greater than the minimum denomination, the coordinator knows those inputs belong to the same user. In an ideal mixing protocol the coordinator should not obtain more information than the already available public ledger data by coordinating the CoinJoin transaction. This information removes many degrees of freedom when assigning non-derived sub-transactions~\cite{maurer2017anonymous}, potentially removing ambiguity when there are multiple valid assignments or reducing the computational cost of such an analysis.
```

Reply:
Just a note: instead of sub-tx it's somewhat equivalent to say sub-mapping (and indeed Maurer et al say that, too.)